--- ./src/main/java/org/apache/maven/plugin/invoker/AbstractInvokerMojo.java.or	2014-12-29 11:01:51.224879274 +0000
+++ ./src/main/java/org/apache/maven/plugin/invoker/AbstractInvokerMojo.java	2014-12-29 11:02:12.724122689 +0000
@@ -2179,7 +2179,7 @@
         try
         {
             Map<String, Object> composite = getInterpolationValueSource();
-            reader = new BufferedReader( new InterpolationFilterReader( newReader( tokenFile ), composite ) );
+            reader = new BufferedReader( new InterpolationFilterReader( newReader( tokenFile ), (Map)composite ) );
 
             String line;
             while ( ( line = reader.readLine() ) != null )
@@ -2242,7 +2242,7 @@
                 // interpolation with token @...@
                 Map<String, Object> composite = getInterpolationValueSource();
                 reader = ReaderFactory.newXmlReader( originalFile );
-                reader = new InterpolationFilterReader( reader, composite, "@", "@" );
+                reader = new InterpolationFilterReader( reader, (Map)composite, "@", "@" );
                 xml = IOUtil.toString( reader );
             }
             finally
